### PR TITLE
Added plugins to monitor number of files and storages

### PR DESCRIPTION
**Pull request by alejandroscf**
This plugins monitor the total number of files in the repository (nextcloud_files) and the number of storages available (nextcloud_storage)

### CODE REVIEW

File: nextcloud_files [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+# Parameters understood:
+#     config   (required)
+#     autoconf (optional - used by munin-config)
+
+# Magic markers - optional - used by installation scripts and
+# munin-config:
+#
+#  #%# family=manual
+#  #%# capabilities=autoconf
+import re
```
Comment (mightyBroccoli) [inline nextcloud_files]:
```suggestion
```
The re import is unnecessary here, as you are not using it.

File: nextcloud_files [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+class NextcloudStorage:
+	def config(self):
+		config = {
+			'storage': [
+				'graph_title Nextcloud Files',
+				'graph_args --base 1000 -l 0',
+				'graph_vlabel number of files',
+				'graph_info graph showing the number of files',
+				'graph_category nextcloud',
+                                'num_files.label number of files',
```
Comment (mightyBroccoli) [inline nextcloud_files]:
```suggestion
				'num_files.label number of files',
```
Please align your indentation to the rest of the dictionary with tab characters and not single spaces, to keep everything consistent.

File: nextcloud_files [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+		return config
+
+	def get_data(self, api_response):
+		data ={
+			'nextcloud_storage': [],
+		}
+
+		# storage
+		storage = api_response['ocs']['data']['nextcloud']['storage']
+		data['nextcloud_storage'].append('multigraph nextcloud_files')
```
Comment (mightyBroccoli) [inline nextcloud_files]:
```suggestion
```
This is not a multigraph plugin. Only if the plugin is outputting multiple graphs, this line is necessary.

File: nextcloud_storage [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+# Parameters understood:
+#     config   (required)
+#     autoconf (optional - used by munin-config)
+
+# Magic markers - optional - used by installation scripts and
+# munin-config:
+#
+#  #%# family=manual
+#  #%# capabilities=autoconf
+import re
```
Comment (mightyBroccoli) [inline nextcloud_storage]:
```suggestion
```
The re import is unnecessary here, as you are not using it.

File: nextcloud_storage [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+		return config
+
+	def get_data(self, api_response):
+		data ={
+			'nextcloud_storage': [],
+		}
+
+		# storage
+		storage = api_response['ocs']['data']['nextcloud']['storage']
+		data['nextcloud_storage'].append('multigraph nextcloud_storage')
```
Comment (mightyBroccoli) [inline nextcloud_storage]:
```suggestion
```
This is not a multigraph plugin. Only if the plugin is outputting multiple graphs, this line is necessary.

**Review by mightyBroccoli — CHANGES_REQUESTED** [review]
Just some small things need to be changed to get this on the way to be merged.

File: nextcloud_files [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+                                'num_files.label number of files',
+                                'num_files.info current number of files in the repository',
+                                'num_files.min 0'
+			]
+		}
+
+		return config
+
+	def get_data(self, api_response):
+		data ={
```
Comment (mightyBroccoli) [inline nextcloud_files]:
```suggestion
		data = {
```
Please add a space after the operator.

File: nextcloud_files [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+			elif sys.argv[1] == 'autoconf':
+				if None in [os.environ.get('username'), os.environ.get('password')]:
+					print('env variables are missing')
+				else:
+					print('yes')
+		else:
+			self.run()
+
+if __name__ == "__main__":
+	NextcloudStorage().main()
```
Comment (mightyBroccoli) [inline nextcloud_files]:
```suggestion
	NextcloudStorage().main()
	
```
PEP8 compliance: please add a newline to the end

**Review by mightyBroccoli — CHANGES_REQUESTED** [review]
some PEP8 compliance changes needed

File: nextcloud_storage [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+				'num_storages_other.label number of other storages',
+				'num_storages_other.info current over all total of storage',
+				'num_storages_other.min 0'
+			]
+		}
+
+		return config
+
+	def get_data(self, api_response):
+		data ={
```
Comment (mightyBroccoli) [inline nextcloud_storage]:
```suggestion
		data = {
```
Please add whitespaces after the operator.

**Review by mightyBroccoli — CHANGES_REQUESTED** [review]
missed one pep8 compliance issue

File: nextcloud_storage [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+			elif sys.argv[1] == 'autoconf':
+				if None in [os.environ.get('username'), os.environ.get('password')]:
+					print('env variables are missing')
+				else:
+					print('yes')
+		else:
+			self.run()
+
+if __name__ == "__main__":
+	NextcloudStorage().main()
```
Comment (mightyBroccoli) [inline nextcloud_storage]:
```suggestion
	NextcloudStorage().main()
	
```
PEP8 please finish with a new line

**Review by mightyBroccoli — CHANGES_REQUESTED** [review]
PLEASE these issues are really just whitespaces please fix them

**Comment by mightyBroccoli** [comment]
@alejandroscf I think I'd rather have all these values ​​fetched by one plugin than two separate ones. They both fetch storage values, which would create a much clearer graph.

But I would be pleased to hear your argumentation on why you choose to use separate ones.

**Comment by alejandroscf** [comment]
Sorry for the delay I was working abroad with no time to fix this errors. Thanks for the corrections.

I decided to use two graph because (at least in my case) num_files outputs a value three orders of magnitude greater than the _storage metrics (3000 vs. 3). Putting everything in the same graph made the _storage values so close to 0 that they were not useful .

**Comment by mightyBroccoli** [comment]
No problem 🦄 
I threw the two plugins together yesterday evening to take a look at the resulting graph and I think you are completely right with your assumption. You can see shit with that big of an difference between the values.
2018-11-13 15_25_20-munin munin __ magicbroccoli de __ rosewood magicbroccoli de __ nextcloud storag

Just some small little changes I merge this pull request.
- the config section in the nextcloud_files script is not correctly indented
- please rename both files to xyz.py ( it seems automated testing is not working if the .py is missing )
- set the x on both files that they are executable, as munin seems to have a problem with python scripts if they are not executable

**Comment by alejandroscf** [comment]
I fixed the tabs, renamed the files and made them executable.

**Pull request merged**